### ISSUE TITLE
implement new search endpoint in Node

### ISSUE THREAD

**Issue by davelandry** [DataUSA/datausa-site#681]
will need the following features replicated from the old endpoint:

- [ ] fuzzy matching
- [ ] fuzzy suggestions
- [ ] filter by type
- [ ] z-scoring
- [ ] search by section/topic

**Comment by davelandry** [comment]
@jspeis wondering what you think the best way to do this will be. I looked at how you do it in CNY, with static lookup files, but I think because of the complexity here with weighting it might make more sense to store it in a database table. We have a separate front-end DB for the CMS, so I think this could all be hooked up and handled in the front-end. I could set up a script to ping specific cube endpoints to calculate the full list with z-score and then populate a table.

**Comment by jspeis** [comment]
Yes, with CNY since it only deals with a subset of geographies within 5 counties in NY, it was definitely easy to work with the static lookups. 

I think the DB approach works as Data USA has substantially more entities than CNY and that can scale as more entities are added. The only thing I would emphasize is to ensure from the get go that the script that will ping the cube endpoints will be able to refresh / "upsert" (update or insert) as new data becomes available from the cubes.
